### PR TITLE
Adds support for Fugue Option to CriteriaModel

### LINKED ISSUE #1161: Fugue Option unsupported by Criteria API
After upgrading a project from `Mongo.Repository` to `Criteria.Repository`, the annotation processor fails at this line.

https://github.com/immutables/immutables/blob/7e736f0f478dcefdc2b815bc869a1a766fc5d068/value-processor/src/org/immutables/value/processor/meta/CriteriaModel.java#L206

It appears that Fugue's `Option` is not handled here. Here's a patch which adds support.

### PR DESCRIPTION
**Pull request by clamothe**
Fixes #1161

### CODE REVIEW

File: criteria/common/pom.xml [hunk 1]
```diff
@@ -33,6 +33,11 @@
       <artifactId>guava</artifactId>
       <version>${guava.version}</version>
     </dependency>
+    <dependency>
```
Comment (asereda-gs) [inline criteria/common/pom.xml]: Is it possible to avoid runtime dependency on fudge ?

**Comment by asereda-gs** [comment]
Thanks for PR. 
Sorry I pushed some changes to `CriteriaModel`. Can you pls rebase with latest master ? 

Although we don't want to support too many optional types this change looks fine because  type can be compared as string (so no direct dependency on atlassian library). 

Btw what backend are you using ?

**Comment by clamothe** [comment]
Thanks. We're using Mongo, albeit haven't yet successfully switched over from `Repository` to `Criteria`.

File: criteria/common/pom.xml (hunk 1 above)
Comment (clamothe) [reply criteria/common/pom.xml]: Done

**Pull request merged**